src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -683,5 +683,69 @@ protected override ComputedInstanceFieldLayout ComputeInstanceFieldLayout(Metada
                 return ComputeAutoFieldLayout(type, numInstanceFields);
             }
         }
+
+        /// <summary>
+        /// This method decides whether the type needs aligned base offset in order to have layout resilient to 
+        /// base class layout changes.
+        /// </summary>
+        protected override void AlignBaseOffsetIfNecessary(MetadataType type, ref LayoutInt baseOffset)
+        {
+            if (type.IsValueType)
+            {
+                return;
+            }
+            DefType baseType = type.BaseType;
+            if (baseType == null || baseType.IsObject)
+            {
+                return;
+            }
+            HashSet<DefType> recursionGuard = new HashSet<DefType>();
+            if (LayoutDependsOnOtherModules(baseType, ((EcmaType)type.GetTypeDefinition()).EcmaModule, recursionGuard))
+            {
+                LayoutInt alignment = new LayoutInt(type.Context.Target.PointerSize);
+                if (type.RequiresAlign8())
+                {
+                    alignment = new LayoutInt(8);
+                }
+                baseOffset = LayoutInt.AlignUp(baseOffset, alignment, type.Context.Target);
+            }
+        }
+
+        private bool LayoutDependsOnOtherModules(DefType type, EcmaModule module, HashSet<DefType> recursionGuard)
```

Comment:
Is field layout algorithm the only place that will need to answer this question?

It kind of feels like this should be in a more central location. I assume "depends on other modules" is the right thing to compute as long as the version bubble === module. Maybe this is going to change depending on the definition of the version bubble?

Reply:
I agree this should be probably rather "LayoutDependsOnOtherVersionBubbles". I'll modify the method like this and I'll move it to MarshalUtils where we already have similar type queries.

Reply:
Fixed in 5th commit by moving the recursive check to ReadyToRunCompilerContext.